java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -408,11 +408,15 @@ public static ArrowRecordBatch deserializeRecordBatch(RecordBatch recordBatchFB,
       ArrowBuf vectorBuffer = body.slice(bufferFB.offset(), bufferFB.length());
       buffers.add(vectorBuffer);
     }
+
+    ArrowBodyCompression bodyCompression =
+        new ArrowBodyCompression(recordBatchFB.compression().codec(), recordBatchFB.compression().method());
```

Comment:
does compression need to be checked for nullness?

Reply:
According to our current implementation. The compression object cannot be null.
For the sake of safety, we added check here.